src/c++/perf_analyzer/command_line_parser.cc

```diff
@@ -0,0 +1,1407 @@
+                   18)
+            << std::endl;
+
+  std::cerr
+      << FormatMessage(
+             " --trace-file: Set the file where trace output will be saved."
+             " If --trace-log-frequency is also specified, this argument "
+             "value will be the prefix of the files to save the trace "
+             "output. See --trace-log-frequency for details. Only used for "
+             "service-kind of triton. Default value is none.",
+             18)
+      << std::endl;
+  std::cerr
+      << FormatMessage(
+             "--trace-level: Specify a trace level. OFF to disable tracing, "
+             "TIMESTAMPS to trace timestamps, TENSORS to trace tensors. It "
+             "may be specified multiple times to trace multiple "
+             "informations. Default is OFF.",
+             18)
+      << std::endl;
+  std::cerr
+      << FormatMessage(
+             " --trace-rate: Set the trace sampling rate. Default is 1000.", 18)
+      << std::endl;
+  std::cerr << FormatMessage(
+                   " --trace-count: Set the number of traces to be sampled. "
+                   "If the value is -1, the number of traces to be sampled "
+                   "will not be limited. Default is -1.",
+                   18)
+            << std::endl;
+  std::cerr
+      << FormatMessage(
+             " --log-frequency:  Set the trace log frequency. If the "
+             "value is 0, Triton will only log the trace output to "
+             "<trace-file> when shutting down. Otherwise, Triton will log "
+             "the trace output to <trace-file>.<idx> when it collects the "
+             "specified number of traces. For example, if the log frequency "
+             "is 100, when Triton collects the 100-th trace, it logs the "
+             "traces to file <trace-file>.0, and when it collects the 200-th "
+             "trace, it logs the 101-th to the 200-th traces to file "
+             "<trace-file>.1. Default is 0.",
+             18)
+      << std::endl;
+
+  std::cerr << FormatMessage(
+                   " --triton-server-directory: The Triton server install "
+                   "path. Required by and only used when C API "
+                   "is used (--service-kind=triton_c_api). "
+                   "eg:--triton-server-directory=/opt/tritonserver.",
+                   18)
+            << std::endl;
+  std::cerr
+      << FormatMessage(
+             " --model-repository: The model repository of which the model is "
+             "loaded. Required by and only used when C API is used "
+             "(--service-kind=triton_c_api). "
+             "eg:--model-repository=/tmp/host/docker-data/model_unit_test.",
+             18)
+      << std::endl;
+  std::cerr << FormatMessage(
+                   " --verbose-csv: The csv files generated by perf analyzer "
+                   "will include additional information.",
+                   18)
+            << std::endl;
+  exit(GENERIC_ERROR);
```

Comment:
Should this call a wrapper function for exit so that this method could be unit tested more easily (i.e. mocking the wrapper)?

Reply:
Never mind, I see the unit test class just overloads the virtual method 👍